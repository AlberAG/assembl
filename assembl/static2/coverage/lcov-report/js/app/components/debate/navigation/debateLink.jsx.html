<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/debate/navigation/debateLink.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">js/app/components/debate/navigation</a> debateLink.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/15</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">63.16% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>12/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">11.11% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/14</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import * as React from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
&nbsp;
import TimelineCpt from './timeline';
import TimelineSegmentMenu from './timelineSegmentMenu';
import { isMobile } from '../../../utils/globalFunctions';
&nbsp;
type DebateLinkProps = {
  timeline: Timeline,
  identifier: string,
  className: string,
  activeClassName: string,
  children: React.Node,
  dataText: string,
  screenTooSmall: boolean
};
&nbsp;
type DebateLinkState = {
  timeLineActive: boolean,
  activeSegment: number
};
&nbsp;
export class DumbDebateLink extends React.Component&lt;DebateLinkProps, DebateLinkState&gt; {
  state = {
    timeLineActive: false,
    activeSegment: -1
  };
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidMount() {
<span class="cstat-no" title="statement not covered" >    document.addEventListener('click', this.handleClickOutside);</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentWillUnmount() {
<span class="cstat-no" title="statement not covered" >    document.removeEventListener('click', this.handleClickOutside);</span>
  }
&nbsp;
  debateNode = null;
&nbsp;
  showMenu = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ timeLineActive: true });</span>
  };
&nbsp;
  hideMenu = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ timeLineActive: false, activeSegment: -1 });</span>
  };
&nbsp;
  handleClickOutside = <span class="fstat-no" title="function not covered" >(e</span>vent: MouseEvent) =&gt; {
    // Cannot call `this.debateNode.contains` with `event.target` bound to `other`
    // because `EventTarget` [1] is incompatible with `Node`
    // $FlowFixMe
<span class="cstat-no" title="statement not covered" >    if (this.state.timeLineActive &amp;&amp; this.debateNode &amp;&amp; !this.debateNode.contains(event.target)) {</span>
<span class="cstat-no" title="statement not covered" >      this.hideMenu();</span>
    }
  };
&nbsp;
  showSegmentMenu = <span class="fstat-no" title="function not covered" >(i</span>ndex: number) =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({ activeSegment: index });</span>
  };
&nbsp;
  render() {
    const { identifier, children, className, activeClassName, dataText, screenTooSmall, timeline } = this.props;
    const { timeLineActive, activeSegment } = this.state;
    // timeline can still be null (loading) if Sections query returned before Timeline query
    const activeSegmentPhase = timeline ? timeline[activeSegment] : <span class="branch-1 cbranch-no" title="branch not covered" >undefined;</span>
    // The first touch show the menu and the second activate the link
    const isTouchScreenDevice = isMobile.any();
    return (
      &lt;div
        ref={<span class="fstat-no" title="function not covered" >(d</span>ebateNode) =&gt; {
<span class="cstat-no" title="statement not covered" >          this.debateNode = debateNode;</span>
        }}
        className={classNames('debate-link', { active: timeLineActive })}
        onMouseOver={!isTouchScreenDevice &amp;&amp; !screenTooSmall ? this.showMenu : null}
        onMouseLeave={!isTouchScreenDevice &amp;&amp; !screenTooSmall ? this.hideMenu : null}
      &gt;
        &lt;div className={classNames(className, activeClassName)} data-text={dataText}&gt;
          {children}
        &lt;/div&gt;
        {!screenTooSmall &amp;&amp; (
          &lt;div className="header-container"&gt;
            &lt;section className="timeline-section" id="timeline"&gt;
              &lt;div className="max-container"&gt;
                &lt;TimelineCpt
                  timeline={timeline}
                  activeSegment={activeSegment}
                  showNavigation
                  identifier={identifier}
                  onItemSelect={this.showSegmentMenu}
                  onItemDeselect={this.hideMenu}
                /&gt;
              &lt;/div&gt;
            &lt;/section&gt;
            {activeSegmentPhase &amp;&amp; (
<span class="branch-1 cbranch-no" title="branch not covered" >              &lt;TimelineSegmentMenu</span>
                phaseIdentifier={activeSegmentPhase.identifier}
                phaseId={activeSegmentPhase.id}
                title={activeSegmentPhase.title}
                onMenuItemClick={this.hideMenu}
                startDate={activeSegmentPhase.start}
                endDate={activeSegmentPhase.end}
              /&gt;
            )}
          &lt;/div&gt;
        )}
      &lt;/div&gt;
    );
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  timeline: state.timeline
});
&nbsp;
export default connect(mapStateToProps)(DumbDebateLink);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
