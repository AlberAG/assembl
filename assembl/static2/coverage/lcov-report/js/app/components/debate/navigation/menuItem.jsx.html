<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/debate/navigation/menuItem.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">js/app/components/debate/navigation</a> menuItem.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.11% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>15/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">53.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/15</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
&nbsp;
import { isMobile } from '../../../utils/globalFunctions';
import { get, goTo } from '../../../utils/routeMap';
import PostsAndContributorsCount from '../../common/postsAndContributorsCount';
&nbsp;
type ItemNode = {
  id: string,
  title: string,
  img: {
    externalUrl: string
  },
  numContributors: number,
  numPosts: number
};
&nbsp;
type MenuItemProps = {
  item: ItemNode,
  identifier: string,
  phaseId: string,
  selected: boolean,
  hasSubItems: boolean,
  slug: string,
  onClick: Function,
  onMouseOver: Function,
  onMouseLeave: Function
};
&nbsp;
export class DumbMenuItem extends React.Component&lt;MenuItemProps&gt; {
  showMenu = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const { onMouseOver, item } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >    if (onMouseOver) <span class="cstat-no" title="statement not covered" >onMouseOver(item.id);</span></span>
  };
&nbsp;
  hideMenu = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const { onMouseLeave, item } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >    if (onMouseLeave) <span class="cstat-no" title="statement not covered" >onMouseLeave(item.id);</span></span>
  };
&nbsp;
  onLinkClick = <span class="fstat-no" title="function not covered" >()</span> =&gt; {
    const { identifier, phaseId, item, slug, onClick } = <span class="cstat-no" title="statement not covered" >this.props;</span>
<span class="cstat-no" title="statement not covered" >    if (onClick) <span class="cstat-no" title="statement not covered" >onClick();</span></span>
<span class="cstat-no" title="statement not covered" >    goTo(get('idea', { slug: slug, phase: identifier, phaseId: phaseId, themeId: item.id }));</span>
  };
&nbsp;
  render() {
    const { item, selected, hasSubItems } = this.props;
    const { title, img, numContributors, numPosts } = item;
    // The first touch show the menu and the second activate the link
    const isTouchScreenDevice = isMobile.any();
    const displayArrow = selected &amp;&amp; hasSubItems;
    const touchActive = isTouchScreenDevice &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >hasSubItems </span>&amp;&amp; <span class="branch-2 cbranch-no" title="branch not covered" >!selected;</span>
    const onLinkClick = touchActive ? <span class="branch-0 cbranch-no" title="branch not covered" >this.showMenu </span>: this.onLinkClick;
    return (
      &lt;div
        className={classNames('menu-item-container', {
          active: selected,
          empty: selected &amp;&amp; !displayArrow
        })}
        onMouseOver={!isTouchScreenDevice &amp;&amp; this.showMenu}
        onMouseLeave={!isTouchScreenDevice &amp;&amp; this.hideMenu}
      &gt;
        &lt;div onClick={onLinkClick} className="menu-item"&gt;
          &lt;div className="thumb-img" style={img &amp;&amp; img.externalUrl ? { backgroundImage: `url(${img.externalUrl})` } : <span class="branch-1 cbranch-no" title="branch not covered" >null}</span>&gt;
            &lt;div className="thumb-img-background" /&gt;
          &lt;/div&gt;
          &lt;div className="thumb-body"&gt;
            &lt;div title={title} className="thumb-title"&gt;
              {title}
            &lt;/div&gt;
            &lt;PostsAndContributorsCount className="menu-stats" numContributors={numContributors} numPosts={numPosts} /&gt;
          &lt;/div&gt;
          {displayArrow &amp;&amp; &lt;span className="thumb-arrow assembl-icon assembl-icon-right-dir" /&gt;}
        &lt;/div&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  slug: state.debate.debateData.slug
});
&nbsp;
export default connect(mapStateToProps)(DumbMenuItem);</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
