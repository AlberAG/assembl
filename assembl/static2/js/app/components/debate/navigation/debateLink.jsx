// @flow
import React from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';

import Timeline from './timeline';
import TimelineSegmentMenu from './timelineSegmentMenu';
import { isMobile } from '../../../utils/globalFunctions';
import { goTo } from '../../../utils/routeMap';
import { type DebateType } from './timelineSegment';

type DebateLinkProps = {
  debate: DebateType,
  identifier: string,
  className: string,
  activeClassName: string,
  children: Array<*>,
  to: string,
  dataText: string,
  screenTooSmall: boolean
};

type DebateLinkState = {
  timeLineActive: boolean,
  activeSegment: number
};

export class DumbDebateLink extends React.Component<*, DebateLinkProps, DebateLinkState> {
  state = {
    timeLineActive: false,
    activeSegment: -1
  };

  componentDidMount() {
    document.addEventListener('click', this.handleClickOutside);
  }

  componentWillUnmount() {
    document.removeEventListener('click', this.handleClickOutside);
  }

  debateNode = null;

  wormholes = null;

  showMenu = () => {
    this.setState({ timeLineActive: true });
  };

  hideMenu = () => {
    // this.setState({ timeLineActive: false, activeSegment: -1 });
  };

  handleClickOutside = (event: MouseEvent) => {
    if (this.state.timeLineActive && this.debateNode && !this.debateNode.contains(event.target)) {
      this.hideMenu();
    }
  };

  onLinkClick = () => {
    this.hideMenu();
    goTo(this.props.to);
  };

  showSegmentMenu = (index: number) => {
    this.setState({ activeSegment: index });
  };

  render() {
    const { identifier, children, to, className, activeClassName, dataText, screenTooSmall, debate } = this.props;
    const { timeLineActive, activeSegment } = this.state;
    const activeSegmentPhase = debate.debateData.timeline[activeSegment];
    // The first touch show the menu and the second activate the link
    const isTouchScreenDevice = isMobile.any();
    const touchActive = isTouchScreenDevice && !timeLineActive;
    const onLinkClick = touchActive ? this.showMenu : this.onLinkClick;
    const linkActive = window.location.pathname === to;
    return (
      <div
        ref={(debateNode) => {
          this.debateNode = debateNode;
        }}
        className={classNames('debate-link', { active: timeLineActive })}
        onMouseOver={!isTouchScreenDevice && !screenTooSmall ? this.showMenu : null}
        onMouseLeave={!isTouchScreenDevice && !screenTooSmall ? this.hideMenu : null}
      >
        <div onClick={onLinkClick} className={classNames(className, { [activeClassName]: linkActive })} data-text={dataText}>
          {children}
        </div>
        {!screenTooSmall && (
          <div className="header-container">
            <section className="timeline-section" id="timeline">
              <div className="max-container">
                <Timeline
                  debate={debate}
                  activeSegment={activeSegment}
                  showNavigation
                  identifier={identifier}
                  onItemSelect={this.showSegmentMenu}
                  onItemDeselect={this.hideMenu}
                />
              </div>
            </section>
            {activeSegmentPhase && (
              <TimelineSegmentMenu
                phaseIdentifier={activeSegmentPhase.identifier}
                title={activeSegmentPhase.title}
                onMenuItemClick={this.hideMenu}
                startDate={activeSegmentPhase.start}
                endDate={activeSegmentPhase.end}
              />
            )}
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  debate: state.debate
});

export default connect(mapStateToProps)(DumbDebateLink);