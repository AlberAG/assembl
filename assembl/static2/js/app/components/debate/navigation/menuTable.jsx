// @flow
import React from 'react';
import type { ApolloClient } from 'react-apollo';

import { SurveyTable, IdeasTable } from './tables';
import AllIdeasQuery from '../../../graphql/AllIdeasQuery.graphql';
import DebateThematicsQuery from '../../../graphql/DebateThematicsQuery.graphql';
import { PHASES } from '../../../constants';

const queries = {
  [PHASES.survey]: DebateThematicsQuery,
  default: AllIdeasQuery
};

export function prefetchMenuQuery(client: ApolloClient, variables: Object) {
  const query = queries[variables.identifier];
  client.query({
    query: query || queries.default,
    variables: variables
  });
}

type MenuTableProps = {
  identifier: string
};

function MenuTable(props: MenuTableProps) {
  switch (props.identifier) {
  case PHASES.survey:
    return <SurveyTable {...props} />;
  case PHASES.voteSession:
    return null;
  default:
    return <IdeasTable {...props} />;
  }
}

export default MenuTable;