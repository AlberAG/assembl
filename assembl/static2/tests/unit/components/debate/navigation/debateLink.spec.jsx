import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { DumbDebateLink } from '../../../../../js/app/components/debate/navigation/debateLink';
import '../../../../helpers/setupTranslations';

describe('DebateLink component', () => {
  it('should match the DebateLink', () => {
    const props = {
      identifier: 'survey',
      children: <span>Debate</span>,
      to: 'https://foo.bar',
      className: 'debate-class',
      activeClassName: 'debate-class-active',
      dataText: 'Debate',
      debate: {
        debateData: {
          timeline: []
        }
      }
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DumbDebateLink {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
  it('should match the DebateLink (small screen)', () => {
    const props = {
      identifier: 'survey',
      children: <span>Debate</span>,
      to: 'https://foo.bar',
      className: 'debate-class',
      activeClassName: 'debate-class-active',
      dataText: 'Debate',
      screenTooSmall: true,
      debate: {
        debateData: {
          timeline: []
        }
      }
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DumbDebateLink {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});