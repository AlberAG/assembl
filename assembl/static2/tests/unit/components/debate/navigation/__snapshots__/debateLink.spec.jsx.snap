// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebateLink component should match the DebateLink (small screen) 1`] = `
<div
  className="debate-link"
  onMouseLeave={false}
  onMouseOver={false}
>
  <div
    className="debate-class"
    data-text="Debate"
    onClick={[Function]}
  >
    <span>
      Debate
    </span>
  </div>
</div>
`;

exports[`DebateLink component should match the DebateLink 1`] = `
<div
  className="debate-link"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="debate-class"
    data-text="Debate"
    onClick={[Function]}
  >
    <span>
      Debate
    </span>
  </div>
  <div
    className="header-container"
  >
    <section
      className="timeline-section"
      id="timeline"
    >
      <div
        className="max-container"
      >
        <Connect(DumbTimeline)
          identifier="survey"
          onMenuItemClick={[Function]}
          showNavigation={true}
        />
      </div>
    </section>
  </div>
</div>
`;
